### DISCUSSION TITLE
Error 403 - Forbidden - The api function is not defined in the allowed functions list

### DISCUSSION THREAD

**Discussion by Raza-DG** [microweber/microweber#903] [Q&A]
I create new module but it's not edit/create and delete it's show 403 - Forbidden Error 403 - Forbidden - The api function is not defined in the allowed functions list
=======================================

<?php only_admin_access(); ?>
<script>
    function delete_whatsapp(id) {
        var are_you_sure = confirm("Are you sure?");
        if (are_you_sure == true) {
            var data = {}
            data.id = id;
            var url = "<?php print api_url('delete_whatsapp'); ?>";
            var post = $.post(url, data);
            post.done(function (data) {
                mw.reload_module("whatsapp");
                mw.reload_module("whatsapp/list");
            });
        }
    }
</script>
-------------------------------------------------------------------
<?php
    $data = \Illuminate\Support\Facades\DB::table('whatsapp')
->orderBy('id','DESC')->get();
?>
---------------------------------------------------------------
<?php if ($data): ?>
    <table width="100%" class="mw-ui-table">
        <thead>
        <tr>
            <th>Name</th>
            <th>Phone</th>
            <th></th>
        </tr>
        </thead>
        <tbody>
        <?php foreach ($data as $item): ?>
            <tr>
                <td><?php print $item->name ?></td>
                <td><?php print $item->phone ?></td>
                <td><a class="mw-ui-btn"
                       href="javascript:delete_whatsapp('<?php print $item->id ?>');">Delete
                    </a>
                </td>
            </tr>
        <?php endforeach; ?>
        </tbody>
    </table>
<?php endif; ?>

----------------

api_expose('delete_whatsapp');
function delete_whatsapp($params)
{
    if (!is_admin()) {
        return;
    }
    if (isset($params['id'])) {
        $table = "whatsapp";
        $id = $params['id'];
        return db_delete($table, $id);
    }
}

**Comment by peter-mw** [comment]
Hi, maybe your module is not loaded. 

You must put the `api_expose` in `functions.php` in the module and you need to reload the modules from this button 

![image](https://user-images.githubusercontent.com/5698247/179193456-2d46cd66-4c7b-42d6-912a-94f545e12b7d.png)

After that the `api_expose` will work.

**Reply by Raza-DG** [reply]
it's auto contain csrf token for any api_expose request.
you look in image. Any other way so please guides as.

![Screenshot_5](https://user-images.githubusercontent.com/88921032/179196612-20d84831-29c5-4086-9b1b-4d837734d9c3.png)

**Reply by Raza-DG** [reply]
> 

No bro it's not working I already reload module

**Reply by peter-mw** [reply]
hi, can you verify that your `functions.php` is loaded by putting some `dd('test')` inside

**Reply by Raza-DG** [reply]
> hi, can you verify that your `functions.php` is loaded by putting some `dd('test')` inside

Thanks bro my file name is function.php so now i rename file is functions.php so it's work fine

Thank you!

**Reply by peter-mw** [reply]
Great :)
